pycbc/events/single.py

```diff
@@ -1,28 +1,51 @@
 """ utilities for assigning FAR to single detector triggers
 """
-from pycbc.events import ranking
+import sys
+import h5py
+import numpy as np
+from pycbc.events import ranking, trigger_fits as fits
 from pycbc.types import MultiDetOptionAction
+from pycbc import conversions as conv
+from pycbc import bin_utils
 
-class LiveSingleFarThreshold(object):
+class LiveSingle(object):
     def __init__(self, ifo,
                  newsnr_threshold=10.0,
-                 reduced_chisq_threshold=5,
+                  reduced_chisq_threshold=5,
                  duration_threshold=0,
-                 fixed_ifar=0):
+                 fit_file=None,
+                 sngl_ifo_est_dist='conservative',
+                 fixed_ifar=None):
+        if sngl_ifo_est_dist not in ['conservative', 'mean', 'fixed']:
+            raise NotImplementedError("Single fitting distribution must be "
+                                      "conservative, mean or fixed")
         self.ifo = ifo
         self.newsnr_threshold = newsnr_threshold
         self.reduced_chisq_threshold = reduced_chisq_threshold
-        self.fixed_ifar = fixed_ifar
         self.duration_threshold = duration_threshold
+        self.sngl_ifo_est_dist = sngl_ifo_est_dist
+        self.fit_file = fit_file
+        self.fixed_ifar = fixed_ifar
 
     @staticmethod
     def insert_args(parser):
         parser.add_argument('--single-newsnr-threshold', nargs='+',
                             type=float, action=MultiDetOptionAction)
         parser.add_argument('--single-reduced-chisq-threshold', nargs='+',
                             type=float, action=MultiDetOptionAction)
-        parser.add_argument('--single-fixed-ifar', nargs='+',
+        parser.add_argument('--single-fixed-ifar', nargs='+', default=None,
                             type=float, action=MultiDetOptionAction)
+        parser.add_argument('--fit-definition-file',
+                            help="File which contains definitons of fit "
+                                 "coefficients and counts for specific "
+                                 "single trigger IFAR fitting.")
+        if '--single-fixed-ifar' in sys.argv:
```

Comment:
I am not sure what the purpose of this code is, but I would not recommend peeking into `sys.argv` from a module. There should be a cleaner way to obtain the desired behavior.

Reply:
Again I've worked around this by simply setting that if single-fixed-ifar is set, then it will override any other options. The sys.argv call was an attempt to have optional defaults depending on whether other options are set